src/deploy-cloudrun.ts

```diff
@@ -165,19 +165,15 @@ export async function run(): Promise<void> {
     }
 
     // set PROJECT ID
-    if (projectId) {
-      await setupGcloud.setProject(projectId);
-    } else if (credentials) {
-      projectId = await setupGcloud.setProjectWithKey(credentials);
-    } else if (process.env.GCLOUD_PROJECT) {
-      await setupGcloud.setProject(process.env.GCLOUD_PROJECT);
+    if (!projectId) {
+      if (credentials) {
+        const key = setupGcloud.parseServiceAccountKey(credentials);
+        projectId = key.project_id;
+      } else if (process.env.GCLOUD_PROJECT) {
+        projectId = process.env.GCLOUD_PROJECT;
+      }
```

Comment:
Should we throw an error if we are not able to find a project ID?

Reply:
No, because it's possible that a previous step actually configured `gcloud` (e.g. someone ran `gcloud config set project`). I don't think we should enumerate all possible places from which a project ID could come, since that is going to be difficult to maintain. 

If no project ID was specified, the gcloud command will run without the --project flag, and gcloud will attempt to resolve the project through its normal resolution path. If it fails, it emits a descriptive error message that says "you should set a project"